modules/electrophysiology_browser/php/sessions.class.inc

```diff
@@ -448,6 +481,591 @@ class Sessions extends \NDB_Page
+                FROM physiological_annotation_file
+                WHERE PhysiologicalFileID=:PFID
+                AND FileType="tsv"',
+            $params
+        );
+        $metadataFileID = $db->pselect(
+            'SELECT AnnotationFileID
+                FROM physiological_annotation_file
+                WHERE PhysiologicalFileID=:PFID
+                AND FileType="json"',
+            $params
+        );
+
+        $params['instance_FID'] = $instanceFileID;
+        $params['metadata_FID'] = $metadataFileID;
+
+        //Get the information for each annotation for the given file
+        $annotationInstance  = $db->pselect(
+            'SELECT * FROM physiological_annotation_instance
+                WHERE AnnotationFileID=:instance_FID',
+            $params
+        );
+        $result['instances'] = $annotationInstance;
+
+        //Get the metadata for the annotations of the given file
+        $annotationMetadata = $db->pselect(
+            'SELECT * FROM physiological_annotation_parameter
+                WHERE AnnotationFileID=:metadata_FID',
+            $params
+        );
+        $result['metadata'] = $annotationMetadata;
+
+        //Get all pre-existing label types
+        $annotationLabels = $db->pselect(
+            'SELECT * FROM physiological_annotation_label',
+            $params
+        );
+        $result['labels'] = $annotationLabels;
+
+        return $result;
+    }
+
+    /**
+     * Updates annotation tables when there is a POST request.
+     * Will add new derivative files if none exist for the given instance.
+     * Will either add new annotations or update existing ones.
+     *
+     * @param ServerRequestInterface $request    POST request from server
+     * @param string                 $outputType Output type of current file
+     *
+     * @return void
+     */
+    function _updateAnnotation(ServerRequestInterface $request, string $outputType)
+    {
+        $values        = $request->getQueryParams();
+        $instance_data = $values['instance'];
+        $user          = \NDB_Factory::singleton()->user();
+        $db            = \NDB_Factory::singleton()->database();
+        $params['SID'] = $this->sessionID;
+
+        $query = 'SELECT
+                      pf.PhysiologicalFileID, pf.FilePath
+                    FROM
+                      physiological_file pf ';
+
+        if ($outputType != 'all_types') {
+            $query        .= 'LEFT JOIN physiological_output_type pot ON ';
+            $query        .= 'pf.PhysiologicalOutputTypeID='
+                . 'pot.PhysiologicalOutputTypeID ';
+            $query        .= 'WHERE SessionID=:SID ';
+            $query        .= 'AND pot.OutputTypeName = :OTN ';
+            $params['OTN'] = $outputType;
+        } else {
+            $query .= "WHERE SessionID=:SID";
+        }
+        $physioFileIDs = $db->pselect(
+            $query,
+            $params
+        );
+
+        if ($this->_hasEditDerivPerms($user)) {
```

Comment:
you are building and running queries above this point which will have no effect (as far as i can see) if the user doesn't have the permission, that's inefficient.

You should consider checking for permission as early as possible and returning without wasting time